EDDI/EDDI.cs

```diff
@@ -1506,9 +1564,12 @@ private bool eventUndocked(UndockedEvent theEvent)
 
         private bool eventTouchdown(TouchdownEvent theEvent)
         {
-            // Only pass on this event if our location is set
+            updateCurrentSystem(theEvent.systemname);
+            updateCurrentStellarBody(theEvent.bodyname, theEvent.systemname, theEvent.systemAddress);
+
+            // If playercontrolled, only pass on this event if our longitude and lattitude are set
             // (if not then this is probably being written prior to a `Location` event))
-            if (theEvent.latitude != null && theEvent.longitude != null)
+            if (theEvent.playercontrolled && theEvent.latitude != null && theEvent.longitude != null)
             {
                 Environment = Constants.ENVIRONMENT_LANDED;
                 Vehicle = theEvent.playercontrolled ? Constants.VEHICLE_SHIP : Constants.VEHICLE_SRV;
```

Comment:
We are in an `if (theEvent.playercontrolled && ...` block here. have we defeated the logic of this statement?

Reply:
Concur. Reverted.